pkg/api/meta/restmapper.go

```diff
@@ -503,6 +503,49 @@ func (m *DefaultRESTMapper) RESTMapping(gk unversioned.GroupKind, versions ...st
 	return retVal, nil
 }
 
+// RESTMappings returns the RESTMappings for the provided group kind in a rough internal preferred order. If no
+// kind is found it will return a NoResourceMatchError.
+func (m *DefaultRESTMapper) RESTMappings(gk unversioned.GroupKind) ([]*RESTMapping, error) {
```

Comment:
Seems like `RESTMapping` should call this function to get the list of possibles and then fish its result from here to avoid duplication.